core/src/main/scala/cats/collections/Diet.scala

```diff
@@ -234,69 +236,67 @@ sealed abstract class Diet[A] {
    * min value in the tree
    */
   def min: Option[A] = this match {
-    case EmptyDiet()                           => None
-    case DietNode(Range(x, _), EmptyDiet(), _) => Some(x)
-    case DietNode(_, l, _)                     => l.min
+    case _: EmptyDiet[_]                           => None
+    case DietNode(Range(x, _), _: EmptyDiet[_], _) => Some(x)
+    case DietNode(_, l, _)                         => l.min
   }
 
   /**
    * max value in the tree
    */
   def max: Option[A] = this match {
-    case EmptyDiet()                           => None
-    case DietNode(Range(_, y), _, EmptyDiet()) => Some(y)
-    case DietNode(_, _, r)                     => r.max
+    case _: EmptyDiet[_]                           => None
+    case DietNode(Range(_, y), _, _: EmptyDiet[_]) => Some(y)
+    case DietNode(_, _, r)                         => r.max
   }
 
   @deprecated("the semantics of this method are unclear and unspecified, avoid!", "0.8.0")
   def map[B: Discrete: Order](f: A => B): Diet[B] = this match {
-    case EmptyDiet() => Diet.empty[B]
-    case DietNode(Range(a, b), l, r) => {
+    case _: EmptyDiet[_] => Diet.empty[B]
+
+    case DietNode(Range(a, b), l, r) =>
       val (lp, rp) = (l.map(f), r.map(f))
 
       val n = lp + f(a) + f(b)
 
       merge(n, rp)
-    }
   }
 
   def foldRight[B](s: Eval[B])(f: (A, Eval[B]) => Eval[B])(implicit enumA: Discrete[A], orderA: Order[A]): Eval[B] =
     this match {
-      case EmptyDiet()         => s
+      case _: EmptyDiet[_]     => s
       case DietNode(rng, l, r) => l.foldRight(rng.foldRight(r.foldRight(s)(f), f))(f)
     }
 
   def foldLeft[B](s: B)(f: (B, A) => B)(implicit enumA: Discrete[A], orderA: Order[A]): B = this match {
-    case EmptyDiet()         => s
+    case _: EmptyDiet[_]     => s
     case DietNode(rng, l, r) => r.foldLeft(rng.foldLeft(l.foldLeft[B](s)(f), f))(f)
   }
 
   def foldLeftRange[B](z: B)(f: (B, Range[A]) => B): B = this match {
-    case EmptyDiet()         => z
+    case _: EmptyDiet[_]     => z
     case DietNode(rng, l, r) => r.foldLeftRange(f(l.foldLeftRange(z)(f), rng))(f)
   }
 
   def foldRightRange[B](z: Eval[B])(f: (Range[A], Eval[B]) => Eval[B]): Eval[B] = this match {
-    case EmptyDiet()         => z
+    case _: EmptyDiet[_]     => z
     case DietNode(rng, l, r) => l.foldRightRange(f(rng, r.foldRightRange(z)(f)))(f)
   }
 
   def toList(implicit discrete: Discrete[A], order: Order[A]): List[A] =
     this match {
-      case EmptyDiet() => Nil
+      case _: EmptyDiet[_] => Nil
+
       case DietNode(rng, l, r) =>
         l.toList ::: rng.toList ::: r.toList
     }
 
   def toIterator: Iterator[Range[A]] = this match {
-    case EmptyDiet() => Iterator.empty
+    case _: EmptyDiet[_] => Iterator.empty
+
     case DietNode(focus, left, right) =>
-      left.toIterator.toStream
-        .append(Seq(focus))
-        .append(right.toIterator)
-        .iterator
+      left.toIterator ++ Iterator.single(focus) ++ right.toIterator
```

Comment:
Actually, I didn't want to optimize it (at least, not in this PR), but I just had to get rid of warnings about `Stream` deprecations. Turned out this is the simplest way I figured out so far.